libs/application-generic/src/usecases/tier-restrictions-validator/tier-restrictions-validator.response.ts

```diff
@@ -0,0 +1,10 @@
+export enum ErrorEnum {
+  TIER_LIMIT_EXCEEDED = 'TIER_LIMIT_EXCEEDED',
+}
+
+type TierValidationError = {
+  error: ErrorEnum;
+  message: string;
+};
+
+export type TierRestrictionsValidatorResponse = TierValidationError[] | null;
```

Comment:
maybe the response here should be a little more future looking, allowing multiple types of validation? of that's what the array is for, usually i find record better here to segment different errors